packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2054,11 +2067,23 @@ function renderRootSync(root: FiberRoot, lanes: Lanes) {
         // function and fork the behavior some other way.
         const unitOfWork = workInProgress;
         const thrownValue = workInProgressThrownValue;
-        workInProgressSuspendedReason = NotSuspended;
-        workInProgressThrownValue = null;
-        unwindSuspendedUnitOfWork(unitOfWork, thrownValue);
-
-        // Continue with the normal work loop.
+        switch (workInProgressSuspendedReason) {
+          case SuspendedOnHydration: {
+            // Selective hydration. An update flowed into a dehydrated tree.
+            // Interrupt the current render so the work loop can switch to the
+            // hydration lane.
+            workInProgress = null;
```

Comment:
A fix for this is added in the PR to add the SyncHydrationLane #25878, because it is only reproducible there.

Reply:
I think it could technically be reproducible if the update expired. So how about we merge both of these changes in the same PR? That way they land as an atomic unit in www.

In other words: don't merge this PR on its own. Let's merge the whole stack in #25878.